app/src/main/java/info/nightscout/androidaps/plugins/constraints/dstHelper/DstHelperPlugin.java

```diff
@@ -43,83 +42,53 @@ public DstHelperPlugin() {
-                    break;
-                }
-                c.add(Calendar.MINUTE, 15);
-
-            }
-        }
-        int minutesLeft = (int) ((offsetDetectedTime / T.mins(1).msecs()));
-        /*log.debug("zoneoffset(minutes):"+zoneOffset/T.mins(1).msecs());
-        log.debug("Start offset: "+offset1/T.mins(1).msecs());
-        log.debug("End offset :" + c.get(Calendar.DST_OFFSET)/T.mins(1).msecs());
-        log.debug("Now is:"+startTimeString);
-        log.debug("Detected in(min): "+(offsetDetectedTime/T.mins(1).msecs()));
-        log.debug("Returning value of: " + minutesLeft); */
-        minutesToChange = minutesLeft;
-        return minutesLeft;
+    public static boolean wasDST(Calendar now) {
+        Calendar ago = (Calendar) now.clone();
+        ago.add(Calendar.HOUR, DISABLE_TIMEFRAME_HOURS);
+        return now.get(Calendar.DST_OFFSET) != ago.get(Calendar.DST_OFFSET);
+    }
 
+    public static boolean willBeDST(Calendar now) {
+        Calendar ago = (Calendar) now.clone();
+        ago.add(Calendar.HOUR, WARN_PRIOR_TIMEFRAME_HOURS);
+        return now.get(Calendar.DST_OFFSET) != ago.get(Calendar.DST_OFFSET);
     }
 
-    //Return false if time to DST change is less than 91 and positive
+    //Return false if time to DST change happened in the last 3 hours.
     @Override
     public Constraint<Boolean> isLoopInvocationAllowed(Constraint<Boolean> value) {
-        try {
-            this.dstTest(Calendar.getInstance());
-        } catch (Exception e) {
-            e.printStackTrace();
+
+        PumpInterface pump = ConfigBuilderPlugin.getPlugin().getActivePump();
+        if (pump == null || pump.canHandleDST()) {
+            log.debug("Pump can handle DST");
+            return value;
         }
-        if (this.minutesToChange <= 90 && minutesToChange > 0 && value.value()) {
-            try {
-                LoopPlugin loopPlugin = LoopPlugin.getPlugin();
-                if (loopPlugin.suspendedTo() == 0L) {
-//                    loopPlugin.suspendTo(System.currentTimeMillis() + minutesToChange * T.mins(1).msecs());
-                    warnUser(Notification.DST_LOOP_DISABLED, MainApp.gs(R.string.dst_loop_disabled_warning));
-                } else
-                    log.debug("Loop already suspended");
-
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-            value.set(false, "DST in 90 minutes or less", this);
-        } else if (minutesToChange <= 24 * T.hours(1).mins() && minutesToChange > 0) {
+
+        Calendar cal = Calendar.getInstance();
+
+        if (willBeDST(cal)) {
             warnUser(Notification.DST_IN_24H, MainApp.gs(R.string.dst_in_24h_warning));
         }
+
+        if (!value.value()) {
+            log.debug("Already not allowed - don't check further");
+            return value;
+        }
+
+        if (wasDST(cal)) {
+            LoopPlugin loopPlugin = LoopPlugin.getPlugin();
+            if (loopPlugin.isSuspended()) {
```

Comment:
I think it should be:
```suggestion
            if (!loopPlugin.isSuspended()) {
```